src/selection.ts

```diff
@@ -82,10 +91,17 @@ type getFragmentSelection<
   Fragments extends { [name: string]: any },
 > = Node extends { kind: Kind.INLINE_FRAGMENT; selectionSet: any }
   ? getSelection<Node['selectionSet']['selections'], Type, Introspection, Fragments>
-  : Node extends { kind: Kind.FRAGMENT_SPREAD; name: any }
+  : Node extends { kind: Kind.FRAGMENT_SPREAD; name: any; directives: any[] }
     ? Node['name']['value'] extends keyof Fragments
       ? Fragments[Node['name']['value']] extends FragmentDefDecorationLike
-        ? makeFragmentRef<Fragments[Node['name']['value']]>
+        ? isUnmaskedFragmentRec<Node['directives']> extends true
+          ? getSelection<
```

Comment:
I think, we can probably replace `$tada.fragmentDef`, and instead of it being set to the definition, set it to either the fragment result type or `makeFragmentRef`.

This way, this part of the code doesn't have to have any knowledge of what it's embedding.

Furthermore, since `$tada.fragmentDef` would then be equivalent to `ResultOf<typeof fragment>`, this part of the code doesn't have to do _any_ work 🎉 

This does require us to place `@mask_disable` on the fragment definition, but we could do:

```graphql
fragment FragName on Type @_noMasking {
  # ...
}
```

This _then_ in turn has the benefit of `parse` only having to check the first directive and filtering the `directives` array on the `FragmentDefinition` nodes it pushes

Reply:
I thought about this and locally defined fragments already work out just fine so would this only be for fragments that we define externally outside of a component? If so this might be a tad limiting 😅 I don't mind tbh just thinking out loud

Reply:
Small note, since I was mistaken,
updating `$tada.fragmentDef`’s type wouldn't be enough, since we only use the definition node via `getFragmentsOfDocumentsRec`.

So, we'd only have to update `getFragmentsOfDocumentsRec` and annotate that. It's a bit messy, since `fragmentId` is “just on there”. But we could remove `$tada.fragmentId` and replace `getFragmentsOfDocumentsRec`’s output to add a new annotation, ad-hoc.

Reply:
@JoviDeCroock
> work out just fine so would this only be for fragments that we define externally outside of a component

yea, see note above, since my implementation path was flawed. We could update `getFragmentsOfDocumentsRec` to actually add the annotation that contains the type (either fragment ref or result type)

I know that this gives a lot more freedom on the spread, but I'd argue it'd have to be on the fragment definition anyway, otherwise `FragmentOf` couldn't switch to giving you the `ResultOf`-type, i.e. if we put it on the spread then the consuming component would have to have knowledge of how to define its input type.

This is of course possible, but, if we think of it as if it manipulated the runtime data, by making this configurable on the spread, there's an inconsistency between what's accepted and what's passed.

Reply:
I think I made it work 😅 it works in the example

Reply:
re. docs, do you think we should write a general “Fragment Composition” page under a “Guides” section and mention it there?

I was thinking that a general “Fragment Composition” next to a future “Typed Documents” page (which would explain `TypedDocumentNode`) could explain fragment composition, masking, and component colocation.

We could also call it “Fragment Colocation” to be fair

Reply:
Hmm, should I refrain from documentation until we get the hollistic topic then?